react-native-git-upgrade/cliEntry.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import shell from 'shelljs';
+import Promise from 'promise';
+import yeoman from 'yeoman-environment';
+import TerminalAdapter from 'yeoman-environment/lib/adapter'
+import log from 'npmlog';
+import rimraf from 'rimraf';
+
+import {
+  checkDeclaredVersion,
+  checkMatchingVersions,
+  checkReactPeerDependency,
+  checkGitAvailable,
+  checkNewVersion
+} from './checks';
+
+log.heading = 'git-upgrade';
+
+/**
+ * Promisify the callback-based shelljs function exec
+ * @param command
+ * @param opts
+ * @returns {Promise}
+ */
+function exec(context, command, forceOutput) {
+  return new Promise((resolve, reject) => {
+    shell.exec(command, {silent: (forceOutput ? !forceOutput : !context.cliArgs.verbose)}, (code, stdout, stderr) => {
+      code
+        ? reject(new Error(`Command '${command}' exited with code ${code}:
+stderr: ${stderr}
+stdout: ${stdout}`))
+        : resolve(stdout);
+    });
+  });
+}
+
+function readProjectDescriptors() {
```

Comment:
Not really, it's more a Java thing, it designates the`pom.xml` in a Maven project: https://maven.apache.org/ref/3.3.9/maven-model/maven.html. That's probably why I use it naturally (I have an old Java background).

It could be confusing with unix "file descriptor" which is totally something else.

How about `readPackageFiles` and the corresponding log: `Read package.json files`?